fsharp-backend/tests/Tests/Serialization.Tests.fs

```diff
@@ -0,0 +1,999 @@
+    let testsFor (dict : Dictionary.T<string, string>) (serializerName : string) =
+      dict
+      |> Dictionary.toList
+      |> List.map (fun (typeName, reason) ->
+        // TODO find a roundtrip test
+        test $"check {serializerName} test files are correct for {typeName}" {
+          // For each type, compare the sample data to the file data
+          SampleData.get typeName serializerName reason
+          |> List.iter (fun (name, actualSerializedData) ->
+            let filename = SampleData.fileNameFor typeName serializerName name
+            let expected = File.readfile Config.Serialization filename
+            Expect.equal actualSerializedData expected "matches")
+        })
+    (testsFor Json.Vanilla.allowedTypes "vanilla")
+    @ (testsFor Json.OCamlCompatible.allowedTypes "ocaml")
+
+  let generateTestFiles () : unit =
+    let generate (dict : Dictionary.T<string, string>) (serializerName : string) =
+      dict
+      |> Dictionary.toList
+      |> List.iter (fun (typeName, reason) ->
+        // For each type, compare the sample data to the file data
+        SampleData.get typeName serializerName reason
+        |> List.iter (fun (name, serializedData) ->
+          let filename = SampleData.fileNameFor typeName serializerName name
+          File.writefile Config.Serialization filename serializedData))
+    generate Json.Vanilla.allowedTypes "vanilla"
+    generate Json.OCamlCompatible.allowedTypes "ocaml"
+
+module CustomSerializersTests =
+
+  type Format =
+    { name : string
+      serializer : PT.Oplist -> byte array
+      deserializer : byte array -> PT.Oplist
+      prettyPrinter : Option<PT.Oplist -> string>
+      prefix : string
+      suffix : string
+      prettyPrinterSuffix : string }
+
+  let formats =
+    [ { name = "BinarySerialization"
+        serializer = BinarySerialization.serializeOplist 0UL
+        deserializer = BinarySerialization.deserializeOplist 0UL
+        prettyPrinter = Some(BinarySerialization.Test.serializeOplistToJson 0UL)
+        prefix = "oplist-binary"
+        suffix = ".bin"
+        prettyPrinterSuffix = ".json" }
+      // DvalReprExternal
+      // - toEnduserReadableTextV0
+      // - toPrettyMachineJsonV1
+      // - toDeveloperReprV0
+      // - unsafeOfUnknownJsonV0
+      // - ofUnknownJsonV1
+      // DvalReprInternalDeprecated
+      // - unsafeDvalOfJsonV1
+      // - unsafeDvalToJsonValueV0
+      // - unsafeDvalToJsonValueV1
+      // - toInternalRoundtrippableV0
+      // - ofIntertnalRoundtrippableJsonV0
+      // - toInternalQueryableV1
+      // - ofInternalQueryableV1
+      // - hash 0
+      // - hash 1
+      // DvalReprInternalNew
+      // - toRoundtrippableJsonV0
+      // - parseRoundtrippableJsonV0
+      // Prelude.Json.Vanilla
+      // Prelude.Json.OCamlCompatible
+      // toStringPairs (LegacyBaseHttpClient)
+      // toStringPairs (HttpClient)
+      // LibJwt LegacySerializer.toYojson
+      // LibJwt LegacySerializer.ofYojson
```

Comment:
is the plan to tackle these in future PRs?

Reply:
Yes.